server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java

```diff
@@ -0,0 +1,303 @@
+
+    final List<Long> toBeDeleted = new ArrayList<>();
+
+    Map<String, PropertyRow> rows = context
+      .prepareSelect("select id, prop_key, resource_id, text_value, created_at from properties")
+      .list(row -> new PropertyRow(
+        row.getLong(1),
+        row.getString(2),
+        row.getLong(3),
+        row.getString(4),
+        row.getLong(5)))
+      .stream()
+      .collect(toMap(PropertyRow::key, Function.identity()));
+
+    toBeDeleted.addAll(migrateGlobalWebhooks(context, rows));
+
+    Map<Long, String> projectsKeys = context
+      .prepareSelect("select id, uuid from projects")
+      .list(row -> new projectFlyweight(
+        row.getLong(1),
+        row.getString(2)))
+      .stream()
+      .collect(toMap(projectFlyweight::id, projectFlyweight::uuid));
+
+    toBeDeleted.addAll(migrateProjectsWebhooks(context, rows, projectsKeys));
+
+    context.prepareUpsert(deleteRequest(toBeDeleted)).execute().commit();
+  }
+
+  private List<Long> migrateProjectsWebhooks(Context context, Map<String, PropertyRow> properties, Map<Long, String> projectsKeys) throws SQLException {
+    List<Long> toBeDeleted = new ArrayList<>();
+    PropertyRow index = properties.get("sonar.webhooks.project");
+    if (index == null) {
+      return emptyList();
+    }
+
+    // can't lambda due to checked exception.
+    for (Webhook webhook : extractProjectWebhooksFrom(properties, index.value().split(","), projectsKeys)) {
+      toBeDeleted.addAll(insert(context, webhook));
+    }
+
+    toBeDeleted.add(index.id());
+    return toBeDeleted;
+  }
+
+  private List<Long> migrateGlobalWebhooks(Context context, Map<String, PropertyRow> properties) throws SQLException {
+
+    List<Long> toBeDeleted = new ArrayList<>();
+    PropertyRow index = properties.get("sonar.webhooks.global");
+    if (index == null) {
+      return emptyList();
+    }
+
+    // can't lambda due to checked exception.
+    for (Webhook webhook : extractGlobalWebhooksFrom(context, properties, index.value().split(","))) {
+      toBeDeleted.addAll(insert(context, webhook));
+    }
+
+    toBeDeleted.add(index.id());
+    return toBeDeleted;
+  }
+
+  private List<Long> insert(Context context, Webhook webhook) throws SQLException {
+
+    if (webhook.isValid()) {
+      context.prepareUpsert("insert into webhooks (uuid, name, url, organization_uuid, project_uuid, created_at, updated_at) values (?, ?, ?, ?, ?, ?, ?)")
+        .setString(1, uuidFactory.create())
+        .setString(2, webhook.name())
+        .setString(3, webhook.url())
+        .setString(4, webhook.organisationUuid())
+        .setString(5, webhook.projectUuid())
+        .setLong(6, webhook.createdAt())
+        .setLong(7, webhook.createdAt())
+        .execute()
+        .commit();
+    } else {
+      LOGGER.info("Unable to migrate inconsistent webhook (entry deleted from PROPERTIES) : " + webhook);
+    }
+
+    return webhook.rowIds();
+  }
+
+  private String deleteRequest(List<Long> toBeDeleted) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "deleteRequest" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10345_manage_webhooks&issues=AWGPwo1FFMFpNJFuf2rk&open=AWGPwo1FFMFpNJFuf2rk)

[ISSUE_KEY]: AWGPwo1FFMFpNJFuf2rk